source/common/tracing/lightstep_tracer_impl.h

```diff
@@ -27,87 +31,81 @@ struct LightstepTracerStats {
   LIGHTSTEP_TRACER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
-class LightStepSpan : public Span {
-public:
-  LightStepSpan(lightstep::Span& span, lightstep::Tracer& tracer);
-
-  // Tracing::Span
-  void finishSpan() override;
-  void setOperation(const std::string& operation) override;
-  void setTag(const std::string& name, const std::string& value) override;
-  void injectContext(Http::HeaderMap& request_headers) override;
-  SpanPtr spawnChild(const Config& config, const std::string& name, SystemTime start_time) override;
-
-  lightstep::SpanContext context() { return span_.context(); }
-
-private:
-  lightstep::Span span_;
-  lightstep::Tracer& tracer_;
-};
-
-typedef std::unique_ptr<LightStepSpan> LightStepSpanPtr;
-
 /**
  * LightStep (http://lightstep.com/) provides tracing capabilities, aggregation, visualization of
  * application trace data.
  *
  * LightStepSink is for flushing data to LightStep collectors.
  */
-class LightStepDriver : public Driver {
+class LightStepDriver : public OpenTracingDriver {
 public:
   LightStepDriver(const Json::Object& config, Upstream::ClusterManager& cluster_manager,
                   Stats::Store& stats, ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime,
-                  std::unique_ptr<lightstep::TracerOptions> options);
+                  std::unique_ptr<lightstep::LightStepTracerOptions>&& options);
 
-  // Tracer::TracingDriver
-  SpanPtr startSpan(const Config& config, Http::HeaderMap& request_headers,
-                    const std::string& operation_name, SystemTime start_time) override;
+  // Tracer::OpenTracingDriver
+  const opentracing::Tracer& tracer() const override;
+  PropagationMode propagationMode() const override { return propagation_mode_; }
 
   Upstream::ClusterManager& clusterManager() { return cm_; }
   Upstream::ClusterInfoConstSharedPtr cluster() { return cluster_; }
   Runtime::Loader& runtime() { return runtime_; }
   LightstepTracerStats& tracerStats() { return tracer_stats_; }
+  void setPropagationMode(PropagationMode propagation_mode) {
```

Comment:
Is there any reason this can't be passed via constructor?

Reply:
I can add it as a constructor argument also. I only put it there for testing to get coverage for the different versions of Inject/Extract.

Reply:
In the tests can you use constructor to construct a new driver? IMO it's cleaner to have this be part of object creation and not settable. If it's really required for testing please call it `setPropagationModeForTest`

Reply:
Changed to use constructor.